src/_pytest/_code/code.py

```diff
@@ -246,10 +246,20 @@ def ishidden(self) -> bool:
 
         Mostly for internal use.
         """
-        f = self.frame
-        tbh = f.f_locals.get(
-            "__tracebackhide__", f.f_globals.get("__tracebackhide__", False)
+        tbh = (
+            False
         )  # type: Union[bool, Callable[[Optional[ExceptionInfo[BaseException]]], bool]]
+        for maybe_ns_dct in (self.frame.f_locals, self.frame.f_globals):
```

Comment:
```suggestion
        for maybe_ns_dct in (self.frame.f_globals, self.frame.f_locals):
```

To preserve the same order as before, where locals overwrite globals? Also we probably could use a test for this behavior, as it seems it is not tested currently.

Reply:
look carefully, the previous code picked from f_locals, and failing that picked from f_globals

Reply:
Ahh sorry, it is a bit late already, missed the `else: break` bit. 👍